File: GeoBroker-Commons/src/test/java/de/hasenburg/geobroker/commons/model/BrokerAreaTest.java
package de.hasenburg.geobroker.commons.model;

import de.hasenburg.geobroker.commons.model.disgb.BrokerArea;
import de.hasenburg.geobroker.commons.model.disgb.BrokerInfo;
import de.hasenburg.geobroker.commons.model.spatial.Geofence;
import de.hasenburg.geobroker.commons.model.spatial.Location;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.Test;

import static org.junit.Assert.*;

@SuppressWarnings({"OptionalGetWithoutIsPresent"})
public class BrokerAreaTest {

	private static final Logger logger = LogManager.getLogger();

	@Test
	public void testSerialize() {
		BrokerInfo brokerInfo = new BrokerInfo("brokerId", "address", 1000);
		BrokerArea brokerArea1 = new BrokerArea(brokerInfo, Geofence.circle(Location.random(), 10));
		KryoSerializer kryo = new KryoSerializer();
		byte[] arr = kryo.write(brokerArea1);
		logger.info(arr);
		BrokerArea brokerArea2 = kryo.read(arr, BrokerArea.class);
		assertEquals(brokerArea1, brokerArea2);
	}

	@Test
	public void testContainsLocation() {
		BrokerInfo brokerInfo = new BrokerInfo("brokerId", "address", 1000);
		Location in = new Location(10, 10);
		Location out = new Location(30, 30);
		BrokerArea brokerArea = new BrokerArea(brokerInfo, Geofence.circle(in, 10));

		assertTrue(brokerArea.containsLocation(in));
		assertFalse(brokerArea.containsLocation(out));
	}

	@Test
	public void testResponsibleBroker() {
		String ownId = "repBroker";
		BrokerInfo responsibleBroker = new BrokerInfo(ownId, "address", 1000);
		BrokerInfo otherBroker = new BrokerInfo("otherBroker", "address", 1000);

		BrokerArea brokerArea = new BrokerArea(responsibleBroker, Geofence.circle(Location.random(), 10));

		// check broker info
		assertEquals(responsibleBroker, brokerArea.getResponsibleBroker());
		assertNotEquals(otherBroker, brokerArea.getResponsibleBroker());

		// use check method
		assertTrue(brokerArea.hasResponsibleBroker(responsibleBroker.getBrokerId()));
		assertFalse(brokerArea.hasResponsibleBroker(otherBroker.getBrokerId()));
	}

	@Test
	public void testWorldResponsibility() {
		BrokerInfo brokerInfo = new BrokerInfo("brokerId", "address", 1000);
		BrokerArea brokerArea = new BrokerArea(brokerInfo, Geofence.world());

		logger.info(brokerArea);

		for (int i = 0; i < 100; i++) {
			assertTrue(brokerArea.containsLocation(Location.random()));
		}
	}

}

File: GeoBroker-Commons/src/test/kotlin/de/hasenburg/geobroker/commons/model/disgb/BrokerAreaTest.kt
package de.hasenburg.geobroker.commons.model.disgb

import de.hasenburg.geobroker.commons.model.spatial.Geofence
import de.hasenburg.geobroker.commons.model.spatial.Location
import org.apache.logging.log4j.LogManager
import org.junit.Assert
import org.junit.Test

private val logger = LogManager.getLogger()

class BrokerAreaTest {

    @Test
    fun testSerialize() {
        val brokerInfo = BrokerInfo("brokerId", "address", 1000)
        val brokerArea1 = BrokerArea(brokerInfo, Geofence.circle(Location.random(), 10.0))

        val json = brokerArea1.toJson()
        logger.info(json)
        val brokerArea2: BrokerArea = json.toBrokerArea()
        Assert.assertEquals(brokerArea1, brokerArea2)
    }

    @Test
    fun testContainsLocation() {
        val brokerInfo = BrokerInfo("brokerId", "address", 1000)
        val `in` = Location(10.0, 10.0)
        val out = Location(30.0, 30.0)
        val brokerArea = BrokerArea(brokerInfo, Geofence.circle(`in`, 10.0))
        Assert.assertTrue(brokerArea.containsLocation(`in`))
        Assert.assertFalse(brokerArea.containsLocation(out))
    }

    @Test
    fun testResponsibleBroker() {
        val ownId = "repBroker"
        val responsibleBroker = BrokerInfo(ownId, "address", 1000)
        val otherBroker = BrokerInfo("otherBroker", "address", 1000)
        val brokerArea = BrokerArea(responsibleBroker,
                Geofence.circle(Location.random(), 10.0))
        // check broker info
        Assert.assertEquals(responsibleBroker, brokerArea.responsibleBroker)
        Assert.assertNotEquals(otherBroker, brokerArea.responsibleBroker)
        // use check method
        Assert.assertTrue(brokerArea.hasResponsibleBroker(responsibleBroker.brokerId))
        Assert.assertFalse(brokerArea.hasResponsibleBroker(otherBroker.brokerId))
    }

    @Test
    fun testWorldResponsibility() {
        val brokerInfo = BrokerInfo("brokerId", "address", 1000)
        val brokerArea = BrokerArea(brokerInfo, Geofence.world())
        logger.info(brokerArea)
        for (i in 0..99) {
            Assert.assertTrue(brokerArea.containsLocation(Location.random()))
        }
    }

}